File: 8_kyu/Find_Maximum_and_Minimum_Values_of_a_List.lua
local solution = {}

-- Return minimum of values in arr.
-- Return nil if arr is empty.
function solution.min(arr)
  if arr == nil or #arr == 0 then
    return nil
  end
  min = arr[1]
  n = #arr
  for i = 1,n do
    if arr[i] < min then
      min = arr[i]
    end
  end
  return min
end

-- Return maximum of values in arr.
-- Return nil if arr is empty.
function solution.max(arr)
  if arr == nil or #arr == 0 then
    return nil
  end
  max = arr[1]
  n = #arr
  for i = 1,n do
    if arr[i] > max then
      max = arr[i]
    end
  end
  return max
end

return solution

File: 8_kyu/Find_Maximum_and_Minimum_Values_of_a_List.rs
// Returns minimum of items in arr.
// Returns 0 if arr is empty.
fn minimum(arr: &[i32]) -> i32 {
    match arr.iter().min() {
        None => 0,
        Some(x) => *x,
    }
}

// Returns maximum of items in arr.
// Returns 0 if arr is empty.
fn maximum(arr: &[i32]) -> i32 {
    match arr.iter().max() {
        None => 0,
        Some(x) => *x,
    }
}
